### ISSUE TITLE
Spring Boot 3 and JavaMelody

### ISSUE THREAD

**Issue by robozb** [javamelody/javamelody#1143]
Does It work with Spring Boot 3?

It seems not:

![image](https://user-images.githubusercontent.com/8656998/217226929-07aff407-afbf-4e3e-acab-f359db0e5270.png)

The /control is my context path.

Thanks: Bela

**Comment by RickDeckard76** [comment]
I confirm the above issue: using spring-boot v3.0.2, javamelody (javamelody-spring-boot-starter v1.91.0) seems to not work (HTTP 404 is returned as in the screenshot above)

**Comment by FeiRoN23** [comment]
Same issue, using java melody 1.91 on a spring boot 3 project.

**Comment by leibf2099** [comment]
migrate package javax to jakarta in 2 jars, and change autoConfiguration to META-INF/spring/org.springframework.boot.autoconfigure.AutoConfigurtoation.imports file for starter jar

**Comment by RickDeckard76** [comment]
> migrate package javax to jakarta in 2 jars, and change autoConfiguration to META-INF/spring/org.springframework.boot.autoconfigure.AutoConfigurtoation.imports file for starter jar

Indeed, with spring 3.x, we all are switching to Tomcat 10.x by default, so the change of packages from javax to jackarta seems to be a necessary step.
However, I am not clear on the "autoConfiguration" configuration change..
In any case, is there a chance for the maintainers to have a new release relatively quickly?

**Comment by robozb** [comment]
> > migrate package javax to jakarta in 2 jars, and change autoConfiguration to META-INF/spring/org.springframework.boot.autoconfigure.AutoConfigurtoation.imports file for starter jar
> 
> Indeed, with spring 3.x, we all are switching to Tomcat 10.x by default, so the change of packages from javax to jakarta seems to be a necessary step. However, I am not clear on the "autoConfiguration" configuration change.. In any case, is there a chance for the maintainers to have a new release relatively quickly?

What I could use instead of JavaMelody was Actuator, Prometheus and Grafana (last ones with Docker), on this way, and that is workable. I know these need much more komplex configuration 

![image](https://user-images.githubusercontent.com/8656998/219091052-8a8f21fb-0515-49a1-9037-73ec63fc0c61.png)

If I remember well, I used these instructions:

- https://www.callicoder.com/spring-boot-actuator/
- https://www.callicoder.com/spring-boot-actuator-metrics-monitoring-dashboard-prometheus-grafana/

Spring project application.yml
```
management:
    endpoints:
        web:
            exposure:
                include: "*"
        health:
            show-details: always   
```
Pom dependency:
```
        <dependency>
            <groupId>org.thymeleaf.extras</groupId>
            <artifactId>thymeleaf-extras-springsecurity6</artifactId>
        </dependency>   
```
Prometheus config:
```
# my global config
global:
  scrape_interval:     15s # Set the scrape interval to every 15 seconds. Default is every 1 minute.
  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.
  # scrape_timeout is set to the global default (10s).

# Load rules once and periodically evaluate them according to the global 'evaluation_interval'.
rule_files:
  # - "first_rules.yml"
  # - "second_rules.yml"

# A scrape configuration containing exactly one endpoint to scrape:
# Here it's Prometheus itself.
scrape_configs:
  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
  - job_name: 'prometheus'
    # metrics_path defaults to '/metrics'
    # scheme defaults to 'http'.
    static_configs:
    - targets: ['127.0.0.1:9090']

  - job_name: 'spring-actuator'
    metrics_path: '/control/actuator/prometheus'
    scrape_interval: 5s
    static_configs:
    - targets: ['192.168.0.10:82']
```

Take care this one:
![image](https://user-images.githubusercontent.com/8656998/219092160-edff07a2-0463-4160-b8b1-ee0a3c50345e.png)

**Comment by robozb** [comment]
Anyway I miss JavaMelody reall much, because of that is much easier way!

**Comment by leibf2099** [comment]
1、use tomcat-jakartaee-migration or other tools to process javamelody-core-1.91.0.jar and javamelody-spring-boot-starter-1.91.0.jar;
2、Starting with Spring Boot3 ,the "org.springframework.boot.autoconfigure.EnableAutoConfiguration" auto-configurations registration from META-INF/spring.factories file is no longer working,
the old writing method spring.factories cannot be used.
for javamelody-spring-boot-starter-1.91.0.jar,the META-INF/spring.factories file must be removed, and create a new META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports file,
and write “net.bull.javamelody.JavaMelodyAutoConfiguration” in the file .
![快照1](https://user-images.githubusercontent.com/7090533/219240364-a25d3a90-df61-4853-9b04-a08fa87cfbec.png)

**Comment by RickDeckard76** [comment]
> 1、use tomcat-jakartaee-migration or other tools to process javamelody-core-1.91.0.jar and javamelody-spring-boot-starter-1.91.0.jar; 2、Starting with Spring Boot3 ,the "org.springframework.boot.autoconfigure.EnableAutoConfiguration" auto-configurations registration from META-INF/spring.factories file is no longer working, the old writing method spring.factories cannot be used. for javamelody-spring-boot-starter-1.91.0.jar,the META-INF/spring.factories file must be removed, and create a new META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports file, and write “net.bull.javamelody.JavaMelodyAutoConfiguration” in the file . ![快照1](https://user-images.githubusercontent.com/7090533/219240364-a25d3a90-df61-4853-9b04-a08fa87cfbec.png)

Clear, thank you very much. In a nutshell, you suggested to the maintainers what could be done right away for the new version 1.92.0 :)

**Comment by evernat** [comment]
yes, migration for jakartaee and spring-boot 3 should start soon

**Comment by evernat** [comment]
it's funny that support of jakartaee and spring-boot 3 is asked here while #1145 asks for support of JBoss 4.0 (J2EE and servlet api 2.4 !) in the same week 😄

**Comment by bafbl** [comment]
> yes, migration for jakartaee and spring-boot 3 should start soon

This is good news! We'd like to make JavaMelody a key part of our operational strategy, but use of SpringBoot 3 is definitely coming very soon, too.

**Comment by evernat** [comment]
see #1151 for review, comments and tests for Jakarta EE 9, Tomcat 10 and Spring-Boot 3

**Comment by evernat** [comment]
Snapshot builds of #1151 available at : http://javamelody.org/downloads/javamelody-spring-boot-starter-2.0.0-SNAPSHOT.jar and http://javamelody.org/downloads/javamelody-core-2.0.0-SNAPSHOT.jar

**Comment by Taha-Di-Nero** [comment]
@evernat  could please tell us when 2.0.0 would be released ??

**Comment by kivan-mih** [comment]
Badly need the release as well

**Comment by evernat** [comment]
@Taha-Di-Nero @kivan-mih nobody has reviewed the pull request 1151 for potential issues. So I plan to release as is.

**Comment by mmoayyed** [comment]
If this is not too much work, do you think you could publish SNAPSHOTs to the Sonatype snapshots repository? This will make it much easier for folks to test the new changes and report feedback.

**Comment by evernat** [comment]
PR merged, tested and fixed.
Released as 2.0.0, available in Maven central repository in an hour or so.
Thanks for your patience.

**Comment by robozb** [comment]
It works well :) I tested
